kimchi/src/circuits/polynomials/foreign_mul/foreign_mul_0.rs

```diff
@@ -0,0 +1,107 @@
+// ForeignMul0 - Foreign field element constraints
+//
+//     Foreign field element F is comprised of three 88-bit limbs L0L1L2
+//
+//     * This circuit gate is used to partially constrain L0 or L1
+//     * The rest of L0 and L1 are constrained by a single ForeignMul2
+//     * It operates on the Curr row
+//
+// Column | Curr
+//      0 | L
+//      1 | plookup Lp0
+//      2 | plookup Lp1
+//      3 | plookup Lp2
+//      4 | plookup Lp3
+//      5 | copy Lp4
+//      6 | copy Lp5
+//      7 | crumb Lc0
+//      8 | crumb Lc1
+//      9 | crumb Lc2
+//     10 | crumb Lc3
+//     11 | crumb Lc4
+//     12 | crumb Lc5
+//     13 | crumb Lc6
+//     14 | crumb Lc7
+
+use std::marker::PhantomData;
+
+use crate::circuits::{
+    argument::{Argument, ArgumentType},
+    expr::{Column, E},
+    gate::{CurrOrNext, GateType},
+    polynomial::COLUMNS,
+};
+use ark_ff::{FftField, Zero};
+use CurrOrNext::*;
+
+use super::constraints::{sublimb_plookup_constraint, two};
+
+#[derive(Default)]
+pub struct ForeignMul0<F>(PhantomData<F>);
+
+impl<F> Argument<F> for ForeignMul0<F>
+where
+    F: FftField,
+{
+    const ARGUMENT_TYPE: ArgumentType = ArgumentType::Gate(GateType::ForeignMul0);
+    const CONSTRAINTS: u32 = 13;
+
+    // Constraints for ForeignMul0
+    //   * Operates on Curr row
+    //   * Range constrain all sublimbs except p4 and p5
+    //   * Constrain that combining all sublimbs equals the limb stored in column 0
+    fn constraints() -> Vec<E<F>> {
+        let w = |i| E::cell(Column::Witness(i), Curr);
+
+        // Row structure
+        //  Column w(i) 0    1       ... 4       5    6    7     ... 14
+        //        Curr  limb plookup ... plookup copy copy crumb ... crumb
+
+        // 1) Apply range constraints on sublimbs
+
+        // Create 4 12-bit plookup range constraints
+        let mut constraints: Vec<E<F>> =
+            (1..5).map(|i| sublimb_plookup_constraint(&w(i))).collect();
+
+        // Create 8 2-bit chunk range constraints
+        constraints.append(
+            &mut (7..COLUMNS)
+                .map(|i| sublimb_plookup_constraint(&w(i)))
```

Comment:
is this a typo? I was expecting this to be a call to `sublimb_crumb_constraint` instead...
I guess this verifies as Ok(()) because these are `zero()` in any case (the TODO inside plookup), so it is normal that it didn't complain before 😅

Reply:
Yes - good spot.  yeah, that's certainly why it verified fine.